src/test/java/com/microsoft/sqlserver/jdbc/connection/TimeoutTest.java

```diff
@@ -29,66 +30,124 @@
+            fail(TestResource.getResource("R_shouldNotConnect"));
+        } catch (Exception e) {
+            assertTrue(e.getMessage().contains(TestResource.getResource("R_tcpipConnectionToHost")));
+            timerEnd = System.currentTimeMillis();
+        }
+
+        verifyTimeout(timerEnd - timerStart, timeout);
+
+        DriverManager.setLoginTimeout(0); // Default to 0 again
     }
 
     @Test
     public void testFailoverInstanceResolution() throws SQLException {
-        long timerStart = 0;
         long timerEnd = 0;
 
-        timerStart = System.currentTimeMillis();
+        long timerStart = System.currentTimeMillis();
         // Try a non existing server and see if the default timeout is 15 seconds
-        try (Connection con = PrepUtil
-                .getConnection("jdbc:sqlserver://" + randomServer + ";databaseName=FailoverDB_abc;failoverPartner="
-                        + randomServer + "\\foo;user=sa;password=pwd;")) {} catch (Exception e) {
+        try (Connection con = PrepUtil.getConnection("jdbc:sqlserver://" + randomServer
+                + ";databaseName=FailoverDB_abc;failoverPartner=" + randomServer + "\\foo;user=sa;password=pwd;")) {
+            fail(TestResource.getResource("R_shouldNotConnect"));
+        } catch (Exception e) {
             assertTrue(e.getMessage().contains(TestResource.getResource("R_tcpipConnectionToHost")));
             timerEnd = System.currentTimeMillis();
         }
-        assertTrue(0 != timerEnd, TestResource.getResource("R_shouldNotConnect"));
 
-        long timeDiff = timerEnd - timerStart;
-        assertTrue(timeDiff > 14000);
+        verifyTimeout(timerEnd - timerStart, defaultTimeout);
     }
 
     @Test
     public void testFOInstanceResolution2() throws SQLException {
-        long timerStart = 0;
         long timerEnd = 0;
 
-        timerStart = System.currentTimeMillis();
-        try (Connection con = PrepUtil
-                .getConnection("jdbc:sqlserver://" + randomServer + "\\fooggg;databaseName=FailoverDB;failoverPartner="
-                        + randomServer + "\\foo;user=sa;password=pwd;")) {} catch (Exception e) {
+        long timerStart = System.currentTimeMillis();
+        try (Connection con = PrepUtil.getConnection("jdbc:sqlserver://" + randomServer
+                + "\\fooggg;databaseName=FailoverDB;failoverPartner=" + randomServer + "\\foo;user=sa;password=pwd;")) {
+            fail(TestResource.getResource("R_shouldNotConnect"));
+        } catch (Exception e) {
             timerEnd = System.currentTimeMillis();
         }
-        assertTrue(0 != timerEnd, TestResource.getResource("R_shouldNotConnect"));
 
-        long timeDiff = timerEnd - timerStart;
-        assertTrue(timeDiff > 14000);
+        verifyTimeout(timerEnd - timerStart, defaultTimeout);
+    }
+
+    private void verifyTimeout(long timeDiff, int timeout) {
```

Comment:
Questionable boundaries for a "pass", should probably revisit this in the future.